### PR TITLE
feature(background-task-tenant-lifecycle): More robust Tenant Lifecycle process

### PR DESCRIPTION
**Pull request by andrewschmidt-a**

### CODE REVIEW

File: src/services/tenant-manager/Services/Tasks/ARMTemplates/dps.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{{
```
Comment (JonathanAsbury-ACS) [inline src/services/tenant-manager/Services/Tasks/ARMTemplates/dps.json:1]: Why the double curly brackets?

**Comment by JonathanAsbury-ACS** [comment]
Andrew,
I'd love to see a demo of this.  
How do we get to the error messages when not running locally?

File: src/services/common/Services/External/AzureManagementClient/IAzureManagementClient.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos.Table;
+using Microsoft.Azure.Management.IotHub.Models;
+using Microsoft.Rest.Azure;
+
+namespace Mmm.Iot.Common.Services.External.TableStorage
```
Comment (joe-bethke) [inline src/services/common/Services/External/AzureManagementClient/IAzureManagementClient.cs:13]: is this purposefully in the `TableStorage` namespace?

Comment (andrewschmidt-a) [reply src/services/common/Services/External/AzureManagementClient/IAzureManagementClient.cs]: Nope moved it over in the latest commit, thanks!

**Comment by andrewschmidt-a** [comment]
Do Not Merge, still in progress!

**Comment by joe-bethke** [comment]
I would like to see some unit tests come with this, or a future PR.

**Pull request merged**